server/src/com/thoughtworks/go/server/service/materials/PluggableSCMMaterialPoller.java

```diff
@@ -74,6 +96,19 @@ private void populateConfiguration(Configuration configuration, com.thoughtworks
         }
     }
 
+    private void updateAdditionalData(final long materialId, final Map<String, String> materialData) {
```

Comment:
:( I think everything that reads or write from material-related tables, should be somewhere in the MaterialRepository or around it.

It's your call, finally.